internal/paths/paths.go

```diff
@@ -38,6 +41,9 @@ var stateFileFunc func() string
 // tailscaled state file, or the empty string if there's no reasonable
 // default value.
 func DefaultTailscaledStateFile() string {
+	if state := os.Getenv("TS_STATE_DIR"); state != "" {
+		return state
```

Comment:
This func needs to return the full path to the state file, not just the directory.  Also, to make clearer that the env variable is the directory, not the file, I'd rename the local var to `statedir`:

``` go
if statedir := os.Getenv("TS_STATE_DIR"); statedir != "" {
    return filepath.Join(statedir, "tailscaled.state")
}
```

(same as [what tailscaled does](https://github.com/tailscale/tailscale/blob/e8b06b22323668b28ac4f950b8b8963922d4f8bf/cmd/tailscaled/tailscaled.go#L280))

Reply:
Oh you're right, sorry for that and thanks for fixing and merging it 🙏 !